react/features/youtube-player/components/native/YoutubeLargeVideo.js

```diff
@@ -0,0 +1,233 @@
+
+    /**
+     * Is the video shared by the local user.
+     *
+     * @private
+     */
+    isOwner: Boolean,
+
+    /**
+     * The width of the screen.
+     *
+     * @private
+     */
+    screenWidth: number,
+
+    /**
+     * The height of the screen.
+     *
+     * @private
+     */
+    screenHeight: number,
+
+    /**
+     * True if in landscape mode.
+     *
+     * @private
+     */
+    isWideScreen: Boolean,
+
+    /**
+     * Callback to invoke when { @code isWideScreen} changes.
+     *
+     * @private
+     */
+    onWideScreenChanged: Function
+};
+
+const YoutubeLargeVideo = (props: Props) => {
+    const playerRef = useRef(null);
+
+    useEffect(() => {
+        if (!props.isOwner) {
+            playerRef.current && playerRef.current.getCurrentTime().then(t => {
+                if (props.seek - t > 0.5) {
+                    playerRef.current && playerRef.current.seekTo(props.seek);
+                }
+            });
+        }
+    }, [ props.seek ]);
+
+    useEffect(() => {
+        props.onWideScreenChanged(props.isWideScreen);
+    }, [ props.isWideScreen ]);
+
+    const onChangeState = e => props.onVideoChangeEvent(e, playerRef.current && playerRef.current.getCurrentTime());
+    const onReady = () => props.onVideoReady(playerRef.current && playerRef.current.getCurrentTime());
+
+    let playerHeight, playerWidth;
+
+    if (props.isWideScreen) {
+        playerHeight = props.screenHeight;
+        playerWidth = playerHeight * 16 / 9;
+    } else {
+        playerWidth = props.screenWidth;
+        playerHeight = playerWidth * 9 / 16;
+    }
+
+    return (<View style = { styles.youtubeVideoContainer } >
+        <YoutubePlayer
+            height = { playerHeight }
+            initialPlayerParams = {{
+                controls: props.enableControls,
+                modestbranding: true
+            }}
+            /* eslint-disable react/jsx-no-bind */
+            onChangeState = { onChangeState }
+            /* eslint-disable react/jsx-no-bind */
+            onReady = { onReady }
+            play = { props.isPlaying }
+            playbackRate = { 1 }
+            ref = { playerRef }
+            videoId = { props.youtubeId }
+            volume = { 50 }
+            webViewProps = {{
+                bounces: false,
+                scrollEnabled: false
+            }}
+            width = { playerWidth } />
+    </View>);
+};
+
+/**
+ * Maps (parts of) the Redux state to the associated YoutubeLargeVideo's props.
+ *
+ * @param {Object} state - Redux state.
+ * @private
+ * @returns {Props}
+ */
+function _mapStateToProps(state) {
+    const { ownerId, status, time } = state['features/youtube-player'];
+    const localParticipant = getLocalParticipant(state);
+    const responsiveUi = state['features/base/responsive-ui'];
+    const screenHeight = responsiveUi.clientHeight;
+    const screenWidth = responsiveUi.clientWidth;
+
+    return {
+        isPlaying: status === 'playing',
```

Comment:
We have the convention of prefixing the props injected by mapStateToProps with and `_`